src/lib.rs

```diff
@@ -215,29 +215,29 @@ impl Quaint {
 
                 (manager, connection_limit as u32)
             }
-            _ => unimplemented!(
-                "Supported url schemes: file or sqlite, mysql, postgres or postgresql."
-            ),
+            _ => unimplemented!("Supported url schemes: file or sqlite, mysql, postgres or postgresql."),
         };
 
         let connection_info = ConnectionInfo::from_url(url_str)?;
         Self::log_start(connection_info.sql_family(), connection_limit);
 
-        Ok(Self {
-            inner: Builder::new().build(connection_limit as usize, manager),
-            connection_info,
-        })
+        let inner = Pool::builder()
+            .max_size(connection_limit)
+            .build(manager)
+            .await?;
+
+        Ok(Self { inner, connection_info })
     }
 
     /// The number of connections in the pool.
-    pub fn capacity(&self) -> usize {
-        self.inner.capacity()
+    pub async fn capacity(&self) -> u32 {
+        self.inner.state().await.connections
```

Comment:
it feels a bit weird that you have to await to get the pool's internal state, but it's fine

Reply:
The state is blocked by an async mutex.

Reply:
makes sense